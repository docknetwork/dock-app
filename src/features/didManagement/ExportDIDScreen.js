import React from 'react';
import {addTestId} from '../../core/automation-utils';
import {
  BackButton,
  Content,
  Footer,
  Header,
  Input,
  LoadingButton,
  NBox as Box,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {FormControl, Stack} from 'native-base';
import {translate} from '../../locales';
import {CheckCircle} from '../../components/CheckCircleComponent';

export function ExportDIDScreen() {
  return (
    <ScreenContainer {...addTestId('ExportDIDScreen')}>
      <Header>
        <BackButton {...addTestId('BackButton')} />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography variant="h1" marginTop={52}>
          {translate('didManagement.create_password')}
        </Typography>
        <Typography marginTop={3}>
          {translate('didManagement.create_password_desc')}
        </Typography>

        <Box>
          <FormControl>
            <Stack mt={7}>
              <FormControl.Label>
                {translate('didManagement.enter_password')}
              </FormControl.Label>
              <Input
                placeholder="Password"
                {...addTestId('Password')}
                value={''}
                onChangeText={() => {}}
                autoCapitalize="none"
                secureTextEntry={true}
              />
            </Stack>
          </FormControl>
        </Box>

        <Box mb={7}>
          <FormControl>
            <Stack mt={7}>
              <FormControl.Label>
                {translate('didManagement.confirm_password')}
              </FormControl.Label>
              <Input
                placeholder="Confirm password"
                {...addTestId('ConfirmPassword')}
                value={''}
                onChangeText={() => {}}
                autoCapitalize="none"
                secureTextEntry={true}
              />
            </Stack>
          </FormControl>
        </Box>
        <Stack marginTop={4}>
          <Stack direction="row">
            <CheckCircle checked={true} />
            <Typography>
              {translate('create_password.include_char_length')}
            </Typography>
          </Stack>
          <Stack direction="row" marginTop={3}>
            <CheckCircle checked={false} />
            <Typography>
              {translate('create_password.include_digits')}
            </Typography>
          </Stack>
          <Stack direction="row" marginTop={3}>
            <CheckCircle checked={true} />
            <Typography>
              {translate('create_password.include_proper_case')}
            </Typography>
          </Stack>
          <Stack direction="row" marginTop={3}>
            <CheckCircle checked={false} />
            <Typography>
              {translate('create_password.include_special_characters')}
            </Typography>
          </Stack>
          <Stack direction="row" marginTop={3}>
            <CheckCircle checked={true} />
            <Typography>
              {translate('create_password.passwords_match')}
            </Typography>
          </Stack>
        </Stack>
      </Content>
      <Footer marginBottom={10} marginLeft={26} marginRight={26}>
        <LoadingButton
          full
          {...addTestId('NextBtn')}
          isDisabled={false}
          onPress={() => {}}>
          {translate('navigation.next')}
        </LoadingButton>
      </Footer>
    </ScreenContainer>
  );
}
export function ExportDIDScreenContainer() {
  return <ExportDIDScreen />;
}
