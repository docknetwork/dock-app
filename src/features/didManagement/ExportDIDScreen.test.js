import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ExportDIDScreen} from './ExportDIDScreen';

const mockStore = configureMockStore();

describe('ExportDIDScreen', () => {
  it('should render correctly ', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<ExportDIDScreen />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
