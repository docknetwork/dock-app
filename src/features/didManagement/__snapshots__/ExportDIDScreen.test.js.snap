// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportDIDScreen should render correctly  1`] = `
<RNCSafeAreaView
  accessibilityLabel="ExportDIDScreen"
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
  testID="ExportDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <BackButton
          accessibilityLabel="BackButton"
          testID="BackButton"
        />
      </Styled(View)>
      <Styled(ScrollView)
        marginLeft={26}
        marginRight={26}
      >
        <Styled(Component)
          marginTop={52}
          variant="h1"
        >
          Create a password
        </Styled(Component)>
        <Styled(Component)
          marginTop={3}
        >
          This will be used to import your DID to a new device
        </Styled(Component)>
        <Memo(ForwardRef(Box))>
          <Memo(ForwardRef(FormControl))>
            <Memo(ForwardRef(Stack))
              mt={7}
            >
              <Memo(ForwardRef(FormControlLabel))>
                Enter Password
              </Memo(ForwardRef(FormControlLabel))>
              <Memo(ForwardRef(Input))
                accessibilityLabel="Password"
                autoCapitalize="none"
                onChangeText={[Function]}
                placeholder="Password"
                secureTextEntry={true}
                testID="Password"
                value=""
              />
            </Memo(ForwardRef(Stack))>
          </Memo(ForwardRef(FormControl))>
        </Memo(ForwardRef(Box))>
        <Memo(ForwardRef(Box))
          mb={7}
        >
          <Memo(ForwardRef(FormControl))>
            <Memo(ForwardRef(Stack))
              mt={7}
            >
              <Memo(ForwardRef(FormControlLabel))>
                Confirm Password
              </Memo(ForwardRef(FormControlLabel))>
              <Memo(ForwardRef(Input))
                accessibilityLabel="ConfirmPassword"
                autoCapitalize="none"
                onChangeText={[Function]}
                placeholder="Confirm password"
                secureTextEntry={true}
                testID="ConfirmPassword"
                value=""
              />
            </Memo(ForwardRef(Stack))>
          </Memo(ForwardRef(FormControl))>
        </Memo(ForwardRef(Box))>
        <Memo(ForwardRef(Stack))
          marginTop={4}
        >
          <Memo(ForwardRef(Stack))
            direction="row"
          >
            <CheckCircle
              checked={true}
            />
            <Styled(Component)>
              Must include at least 8 characters
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle
              checked={false}
            />
            <Styled(Component)>
              Must include digits
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle
              checked={true}
            />
            <Styled(Component)>
              Must include uppercase and lowercase letters
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle
              checked={false}
            />
            <Styled(Component)>
              Must include special characters
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle
              checked={true}
            />
            <Styled(Component)>
              Password confirmation match
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
        </Memo(ForwardRef(Stack))>
      </Styled(ScrollView)>
      <Styled(Component)
        marginBottom={10}
        marginLeft={26}
        marginRight={26}
      >
        <LoadingButton
          accessibilityLabel="NextBtn"
          full={true}
          isDisabled={false}
          onPress={[Function]}
          testID="NextBtn"
        >
          Next
        </LoadingButton>
      </Styled(Component)>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
